packages/instantsearch.js/src/middlewares/createInsightsMiddleware.ts

```diff
@@ -138,12 +139,17 @@ export function createInsightsMiddleware<
       // Otherwise, the `init` call might override it with anonymous user token.
       userTokenBeforeInit = userToken;
     });
-    insightsClient('init', {
-      appId,
-      apiKey,
-      useCookie: true,
-      ...insightsInitParams,
-    });
+
+    // Only `init` if user provided the `insights` middleware.
+    if (!$$internal) {
```

Comment:
```suggestion
    if (!$$internal || insightsInitParams) {
```

this would require another test, but you can pass the init params to the insights option, so in that case we should still init.

Reply:
Good catch, thanks Haroen, I added a test and updated the PR description.